src/Checkout/Customer/Manager.php

```diff
@@ -119,13 +119,67 @@ public function createCustomerFromQuote(
 
         $websiteId  = $this->storeManager->getWebsite()->getId();
 
+        $dateOfBirth = $this->extractDateOfBirthFromQuote($quote);
+        if (!$dateOfBirth instanceof \DateTime) {
+            $dateOfBirth = null;
+        } else {
+            $dateOfBirth = $dateOfBirth->format('Y-m-d');
+        }
+
         $customer->setWebsiteId($websiteId)
             ->setLastname($quote->getCustomerLastname())
             ->setFirstname($quote->getCustomerFirstname())
-            ->setEmail($email);
+            ->setEmail($email)
+            ->setDob($dateOfBirth);
+
         return $this->accountManagement->createAccount($customer);
     }
 
+    protected function extractDateOfBirthFromQuote($quote)
+    {
+        if ($quote->getData('collectorbank_customer_type') != 1) {
+
+            return false;
+        }
+        $data = json_decode($quote->getCollectorbankData(), true);
+        $ssn = $data['national_identification_number'] ?? false;
+        $format = $this->detectSsnFormat($quote);
+        return ($ssn && $format)
+            ? $this->normalizeDateFromFormat($format, $ssn)
+            : false;
+    }
+
+    protected function normalizeDateFromFormat($format, $ssn)
+    {
+        switch($format) {
+            case 'ÅÅMMDD-NNNN':
+                $datePart = mb_substr($ssn, 0, 8);
+                return \DateTime::createFromFormat('Ymd', $datePart);
+            case 'DDMMÅÅSNNNK':
+            case 'DDMMÅÅNNNNN':
+                $datePart = mb_substr($ssn, 0, 6);
```

Comment:
When the customer year of birth is < 1970 and we only have two numerals to detect age, this may cause it to think you are born in the future
 
Perhaps we should do something like:
 if it thinks you are young (lets say < 10 years of age)  we assume it's in the 1900's